indra/newview/llviewerwindow.cpp

```diff
@@ -1331,7 +1331,7 @@ LLWindowCallbacks::DragNDropResult LLViewerWindow::handleDragNDrop( LLWindow *wi
                                 // Check the whitelist, if there's media (otherwise just show it)
                                 if (te->getMediaData() == NULL || te->getMediaData()->checkCandidateUrl(url))
                                 {
-                                    if ( obj != mDragHoveredObject)
+                                    if (obj != mDragHoveredObject.get())
```

Comment:
Let's pull in Ansariel's free functions to compare `(T* == LLPointer<T>)` and `!=`.

Reply:
They are already in the `develop` branch (d91d39fa0f7f4f204d6fb7ff66b9817e498dbd61). The problem here is that obj is a LLVOVolume pointer and mDragHoveredObject is an LLPointer<LLViewerObject>, so the overloaded operator doesn't work.

Reply:
If we're comparing pointers to different types in the same class hierarchy, shouldn't we formally be downcasting one of them to match the other? We don't use a lot of multiple inheritance, but a pointer to object A's base class doesn't necessarily have the same bit pattern as a pointer to A.

Reply:
That would probably be a smart thing to do.

Reply:
I should have realized [the language takes care of that](https://en.cppreference.com/w/cpp/language/implicit_conversion#Pointer_conversions).

That being the case, I'll try tweaking the `(T* == LLPointer<T>)` comparisons to accept a raw pointer to arbitrary type, letting the compiler produce an error if the two pointer types can't be compared.

Reply:
[#2681](https://github.com/secondlife/viewer/pull/2681)